cns/restserver/restserver.go

```diff
@@ -280,6 +280,8 @@ func (service *HTTPRestService) Init(config *common.ServiceConfig) error {
 	listener.AddHandler(cns.NetworkContainersURLPath, service.getOrRefreshNetworkContainers)
 	listener.AddHandler(cns.GetHomeAz, service.getHomeAz)
 	listener.AddHandler(cns.EndpointPath, service.EndpointHandlerAPI)
+	listener.AddHandler(cns.GetVMUniqueID, service.getVMUniqueID)
```

Comment:
I don't see risk of this data, specifically, but I am generally concerned with CNS posing as an IMDS proxy. I think this is a good opportunity for us to evaluate which endpoints need to be always enabled, or should only be enabled based on config.

Reply:
This is needed for Swift v2 (for now). I don't think we have any explicit config in CNS to indicate Swift v2 mode

Reply:
Is this endpoint covered under the standalone DNC-CNS mTLS config?
My concern is that if this is for DNC, it is necessarily externally accessible. Exposing IMDS is not a good idea, and I don't want this going to AKS. 
So how can we restrict those 1P implicitly privileged endpoints?

Reply:
This is not part of mTLS. For AKS, is CNS not local-host only?

Reply:
Thinking little bit more about it, there are two options
1. Like Evans said, we add endpoints based on config, which means new endpoints will need a feature/scenario config in CNS and clients will be burdened with enabling it.
2. This is protected via API authentication from DNC to CNS. (which is where mTLS comes into picture). This is the ideal scenario in my mind.

I still don't understand the full nature of concern here and how it impacts AKS-SWIFT (as I am under the impression that we have a local host only endpoint for AKS-SWIFT scenario).

Reply:
> For AKS, is CNS not local-host only?

Not necessarily 🙁  That's why we have added a [separate server recently](https://github.com/Azure/azure-container-networking/pull/2650) - to move the AKS specific endpoints to a `localhost` only access. But afaik this server that you are adding paths to is still started in Swift. Maybe that's the change we should make?

Reply:
Wonder what @paulyufan2 thinks here

Reply:
ok. My suggestion would be to solve this via 
1. local host only endpoints for APIs that don't need to be on the network
2. Authentication between DNC and CNS
3. Also block containers from reaching infra services on host namespace via ACLs or iptable/HNS rules etc.

That is better way to go than each individual developer worrying about adding a new API in CNS. #1 and #2 are already in progress. So this is good but it is long term and will take time.

For short term, only option I see if put it behind a config but we already have CNS APIs that reach out to NMA via wireserver. So not sure why would IMDS specifically be of concern.

Reply:
Evans and I had offline conversation. After getting more information about the AKS security issue, this API is being exposed only for direct channel mode. It doesn't seem like that this API poses any security risk for AKS but I decided to expose it only for direct channel mode since that is the only scenario this API is needed in. In the fullness of time, AKS should move to local host only CNS for any risk from network APIs. Other scenarios should enable mTLS for CNS .